app/lib/actions/katello/repository/destroy.rb

```diff
@@ -29,7 +31,9 @@ def plan(repository, options = {})
             if repository.redhat?
               handle_redhat_content(repository) unless skip_environment_update
             else
-              handle_custom_content(repository) unless skip_environment_update
+              if destroy_content
+                handle_custom_content(repository) unless skip_environment_update
```

Comment:
we should probably combine those statemetns:

if destroy_content && !skip_environment_update